ninja/yaml.py

```diff
@@ -0,0 +1,69 @@
+from collections import OrderedDict
+from decimal import Decimal
+from typing import IO, Mapping, Optional, Tuple
+
+import yaml
```

Comment:
Wrap this in a try except

Reply:
I tried with try/except before, but realized that the rest of the module makes no sense if `yaml` cannot be imported. What should we do in the `except` clause?

Reply:
Lets assume someone wants to access this internal module. But he doesnt have `PyYAML` installed. 

So i think we should remove try except from the parser and move this logic into here. With same error message.